lib/processor.spec.js

```diff
@@ -0,0 +1,203 @@
+import test from 'ava'
+import proxyquire from 'proxyquire'
+import contentful from 'contentful'
+
+test('processor.processFile - return the initial file when no contentful meta is set', t => {
+  const processor = proxyquire(
+    './processor',
+    {
+      contentful
+    }
+  )
+
+  const file = {}
+
+  t.is(processor.processFile(file), true)
+})
+
+test.cb('processor.processFile - call entries with correct properties coming from file meta', t => {
+  const processor = proxyquire(
+    './processor',
+    {
+      contentful: {
+        createClient: function (options) {
+          t.is(options.accessToken, 'bar')
+          t.is(options.host, 'baz')
+          t.is(options.space, 'foo')
+
+          t.end()
+
+          return {
+            entries: function () {
+              return new Promise(resolve => resolve([]))
+            }
+          }
+        }
+      }
+    }
+  )
+
+  processor.processFile({
+    contentful: {
+      host: 'baz',
+      space_id: 'foo'
+    }
+  }, {
+    access_token: 'bar'
+  })
+})
+
+test.cb('processor.processFile - call entries with correct properties coming from global settings', t => {
+  const processor = proxyquire(
+    './processor',
+    {
+      contentful: {
+        createClient: function (options) {
+          t.is(options.accessToken, 'global-bar')
+          t.is(options.host, 'global-baz')
+          t.is(options.space, 'global-foo')
+
+          return {
+            entries: function () {
+              return new Promise(resolve => resolve([]))
+            }
+          }
+        }
+      }
+    }
+  )
+
+  processor.processFile({
+    contentful: {}
+  }, {
+    access_token: 'global-bar',
+    host: 'global-baz',
+    space_id: 'global-foo'
+  }).then(_ => t.end())
```

Comment:
Instead `test.cb` I would just return the Promise as this ensures the test is also finished when the promise is rejected.